### PR TITLE
support num_cpus crate and test that

### PR DESCRIPTION
**Pull request by RalfJung**
Also make some magic numbers into proper global constants.

### CODE REVIEW

File: src/fn_call.rs [hunk 1]
```diff
@@ -622,11 +622,11 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 let name = this.read_scalar(args[0])?.to_i32()?;
 
                 trace!("sysconf() called with name {}", name);
-                // Cache the sysconf integers via Miri's global cache.
```
Comment (RalfJung) [inline src/fn_call.rs]: We don't currently cache this, do we? The comment makes it sound like we do but I found no such code.

**Comment by RalfJung** [comment]
Let's see if this works on Windows. (Let's also see if try builds work. ;)

@bors try

**Comment by RalfJung** [comment]
@rust-lang/infra looks like the try build succeeded but bors didn't post a message about that? Is that expected?
The same happened here: https://github.com/rust-lang/rust/pull/62150#issuecomment-505846859

**Comment by pietroalbini** [comment]
Pretty sure that's because you pushed a commit after the try build started.

**Comment by RalfJung** [comment]
Had to take back the last commit because `getrandom` already uses `getentropy` on macOS, which we don't support yet.

@bors r+

**Pull request merged**